### PR TITLE
Logical Interconnect

### LINKED ISSUE #82: Logical Interconnect
### Scenario/Intent

Implement Logical Interconnect resource and actions.

### PR DESCRIPTION
**Pull request by tmiotto**
### Description

Implements all the Logical Interconnect resource actions
### Issues Resolved
#82
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [x] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: resources/logical_interconnect.rb [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    item = load_resource
+    item.retrieve!
+    current_firmware = item.get_firmware
+    firmware_data['command'] = action
+    dif_values = firmware_data.select { |k, v| current_firmware[k] != v }
+    raise "Unspecified property: 'firmware'. Please set it before attempting this action." unless firmware || firmware_data['sppName']
+    return Chef::Log.info("Firmware #{firmware} from logical interconnect '#{name}' is up to date") if dif_values.empty?
+    fd = OneviewSDK::FirmwareDriver.find_by(item.client, name: firmware).first
+    raise "Resource not found: Firmware action '#{action}' cannot be performed since the firmware '#{firmware}' was not found." unless fd
+    converge_by "#{action.capitalize} firmware '#{firmware}' from logical interconnect '#{name}'" do
```
Comment (jsmartt) [inline resources/logical_interconnect.rb]: Do we want to stick with the pattern and show the diff here?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,9 @@
 ### New
-- Added support to Volume actions `:create_snapshot` and `:delete_snapshot`
-- Added support to SAN manager actions
-- Added support to Uplink set actions
+<<<<<<< HEAD
```
Comment (jsmartt) [inline CHANGELOG.md]: ^

Comment (tmiotto) [reply CHANGELOG.md]: :fearful:

File: resources/logical_interconnect.rb (hunk 1 above)
Comment (tmiotto) [reply resources/logical_interconnect.rb]: We probably want to... I'll fix it!

**Comment by jsmartt** [comment]
@tmiotto , any idea why the Travis builds are failing? I'm confused

**Comment by tmiotto** [comment]
@jsmartt I was really hoping you had any idea why... It's passing on Ruby 2.1 but on the newer versions it is not, because of some `encoding` method, that I have no idea where it's being called. I just know it isn't on my code, unfortunately.

Anyway, I'm trying to figure it out, but it's being a bit hard since it's just happening on Travis.

**Comment by jsmartt** [comment]
I have no clue at this point. Unfortunately I'll be out of the office for a few days and probably won't have time to debug it either :/

**Comment by tmiotto** [comment]
I changed how the unit test is being executed and somehow the error disappeared!

Now it is only failing due to #85

**Comment by hdiomede** [comment]
:+1:

**Pull request merged**